src/codeflare_sdk/cluster/cluster.py

```diff
@@ -200,45 +189,111 @@ def is_ready(self, print_to_console: bool = True):  # pragma: no cover
+        """
+        return f"ray://{self.config.name}-head-svc.{self.config.namespace}.svc:10001"
+
+    def cluster_dashboard_uri(self) -> str:
+        """
+        Returns a string containing the cluster's dashboard URI.
+        """
+        try:
+            with oc.project(self.config.namespace):
+                route = oc.invoke(
+                    "get", ["route", "-o", "jsonpath='{$.items[*].spec.host}'"]
+                )
+                route = route.out().split(" ")
+                route = [x for x in route if f"ray-dashboard-{self.config.name}" in x]
+                route = route[0].strip().strip("'")
+            return f"http://{route}"
+        except:
+            return "Dashboard route not available yet, have you run cluster.up()?"
+
     def list_jobs(self) -> List:
         """
         This method accesses the head ray node in your cluster and lists the running jobs.
         """
-        dashboard_route = self.cluster_dashboard_uri(namespace=self.config.namespace)
+        dashboard_route = self.cluster_dashboard_uri()
         client = JobSubmissionClient(dashboard_route)
         return client.list_jobs()
 
     def job_status(self, job_id: str) -> str:
         """
         This method accesses the head ray node in your cluster and returns the job status for the provided job id.
         """
-        dashboard_route = self.cluster_dashboard_uri(namespace=self.config.namespace)
+        dashboard_route = self.cluster_dashboard_uri()
         client = JobSubmissionClient(dashboard_route)
         return client.get_job_status(job_id)
 
     def job_logs(self, job_id: str) -> str:
         """
         This method accesses the head ray node in your cluster and returns the logs for the provided job id.
         """
-        dashboard_route = self.cluster_dashboard_uri(namespace=self.config.namespace)
+        dashboard_route = self.cluster_dashboard_uri()
         client = JobSubmissionClient(dashboard_route)
         return client.get_job_logs(job_id)
 
 
-def get_current_namespace() -> str:  # pragma: no cover
+def get_current_namespace() -> str:
     """
     Returns the user's current working namespace.
     """
-    namespace = oc.invoke("project", ["-q"]).actions()[0].out.strip()
+    try:
+        namespace = oc.invoke("project", ["-q"]).actions()[0].out.strip()
+    except oc.OpenShiftPythonException as osp:  # pragma: no cover
+        error_msg = osp.result.err()
+        if (
+            "do not have rights" in error_msg
+            or "Missing or incomplete configuration" in error_msg
+        ):
+            raise PermissionError(
+                "Action not permitted, have you run auth.login() or cluster.up()?"
+            )
+        else:
+            raise osp
     return namespace
 
 
-def list_all_clusters(
-    namespace: str, print_to_console: bool = True
-):  # pragma: no cover
+def list_all_clusters(namespace: str, print_to_console: bool = True):
```

Comment:
Should this be `list_all_ray_clusters`?

Reply:
I think it's left generic for "future-proofing", as keeping in line with why we don't mention ray in other user-facing functions, we could theoretically have other types of backends in the future, so it's more so meant to list all "codeflare clusters". Do you think this would lead to any unecessary confusion though? If so, we could change it to `list_all_ray_clusters` for now, or even `list_all_codeflare_clusters`?

Reply:
Ok, that makes sense. I think its fine for now then to keep it general.  I think this is just an artifact of the term "cluster" being a bit overloaded here. Reading through the code, we are basically tracking two types of things, AppWrappers and Ray Clusters. And the term "cluster" seemed somewhat vague. But, something we can maybe address more systematically in the future.